keytree/src/lib.rs

```diff
@@ -20,6 +22,23 @@ impl Xprv {
     }
 }
 
+/// Xpub represents an extended public key.
+pub struct Xpub {
+    point: CompressedRistretto,
+    dk: [u8; 32],
+}
+
+impl Xpub {
+    /// Returns a new Xpub, generated from the provided Xprv.
+    pub fn from_xprv(xprv: Xprv) -> Self {
```

Comment:
suggestion: instead of `Xpub::from_xprv(xprv)` let's turn this around, so it's `xprv.to_xpub()` instead:

```rust
impl Xprv {
    pub fn to_xpub(&self) -> Self {...}
}
```

Also: we don't need to consume xprv to create an xpub, so using a reference (&Xprv in your code, &self in my suggestion) is better.